src/clr-angular/button/button-loading/loading-button.ts

```diff
@@ -5,26 +5,92 @@
  */
 
 
-import {Component} from "@angular/core";
-import {LoadingListener} from "../../utils/loading/loading-listener";
-
+import {animate, keyframes, style, transition, trigger} from "@angular/animations";
+import {Component, ElementRef, EventEmitter, Output, Renderer2} from "@angular/core";
+import {ClrLoadingState, LoadingListener} from "../../utils/loading";
 
 @Component({
     selector: "button[clrLoading]",
     template: `
-        <span class="spinner spinner-inline" *ngIf="loading"></span>
-        <ng-content></ng-content>
+        <ng-container [ngSwitch]="state">
+            <span *ngSwitchCase="buttonState.LOADING">
+                <span @spinner class="spinner spinner-inline spinner-loading"></span>
+            </span>
+            <span *ngSwitchCase="buttonState.SUCCESS">
+                <span @validated class="spinner spinner-inline spinner-loading spinner-check"></span>
+            </span>
+            <span *ngSwitchCase="buttonState.DEFAULT" @defaultButton>
+                <ng-content></ng-content>
+            </span>
+        </ng-container>
     `,
-    providers: [{provide: LoadingListener, useExisting: ClrLoadingButton}]
+    providers: [{provide: LoadingListener, useExisting: ClrLoadingButton}],
+    animations: [
+        trigger("defaultButton",
+                [
```

Comment:
Is this formatting just what clang does automatically? I think its hard to read, if we can do anything about it.

Reply:
Some of it is clang. Let me see if I can improve by at least separating the comment into a new line.